dev: add AI/LLM Assistance section to PR template (#7144)
mealie-recipes/mealie: .github/pull_request_template.md

## Testing

_(fill-in or delete this section)_
<!--

>   Describe how you tested this change.
> -->
> 
> ## AI / LLM Assistance
> <!--
>   Describe to which degree an LLM was used in creating this pull request.
> -->

Reviewer: Can we add "required"?

```suggestion
## AI / LLM Assistance

_(REQUIRED)_

<!--
  Describe to which degree an LLM was used in creating this pull request.
-->
```

Author: Absolutely 👍
